CLI throws an error if download path includes spaces

```
 emindemirci  ~/Downloads/hazelcast-jet-3.0 2  
 ./bin/jet-start.sh 
########################################
# JAVA=/Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home/bin/java
# JAVA_OPTS= -Dhazelcast.config=/Users/emindemirci/Downloads/hazelcast-jet-3.0 2/config/hazelcast.xml -Djet.home=/Users/emindemirci/Downloads/hazelcast-jet-3.0 2 -Dhazelcast.client.config=/Users/emindemirci/Downloads/hazelcast-jet-3.0 2/config/hazelcast-client.xml -Dhazelcast.jet.config=/Users/emindemirci/Downloads/hazelcast-jet-3.0 2/config/hazelcast-jet.xml
# CLASSPATH=/Users/emindemirci/Downloads/hazelcast-jet-3.0 2/lib/hazelcast-jet-3.0.jar:
# starting now....
########################################
Error: Could not find or load main class 2.config.hazelcast.xml
Caused by: java.lang.ClassNotFoundException: 2.config.hazelcast.xml
 1  emindemirci  ~/Downloads/hazelcast-jet-3.0 2  
```

--- a/hazelcast-jet-distribution/src/main/resources/bin/common.sh
+++ b/hazelcast-jet-distribution/src/main/resources/bin/common.sh
@@ -1,15 +1,16 @@
-SCRIPT_DIR=`dirname "$0"`
-JET_HOME=`cd "$SCRIPT_DIR/.." >/dev/null; pwd`
+#!/bin/bash
+SCRIPT_DIR="$(dirname "$0")"
+JET_HOME="$(cd "$SCRIPT_DIR/.."; pwd)"
 PID_FILE=$JET_HOME/bin/jet_instance.pid
 
 if [ $JAVA_HOME ]; then
-    JAVA=$JAVA_HOME/bin/java
+    JAVA="$JAVA_HOME/bin/java"
 else
     echo "JAVA_HOME environment variable not available."
-    JAVA=`which java 2>/dev/null`
+    JAVA="$(which java 2>/dev/null)"
 fi
 
-if [ -z $JAVA ]; then
+if [ -z "$JAVA" ]; then
     echo "Java could not be found in your system."
     exit 1
 fi
@@ -21,33 +22,35 @@ fi
 #### maximum heap size
 # MAX_HEAP_SIZE=1G
 
+declare -a JAVA_OPTS
+
 if [ "x$MIN_HEAP_SIZE" != "x" ]; then
-    JAVA_OPTS="$JAVA_OPTS -Xms${MIN_HEAP_SIZE}"
+    JAVA_OPTS+=("-Xms${MIN_HEAP_SIZE}")
 fi
 
 if [ "x$MAX_HEAP_SIZE" != "x" ]; then
-    JAVA_OPTS="$JAVA_OPTS -Xmx${MAX_HEAP_SIZE}"
+    JAVA_OPTS+=("-Xmx${MAX_HEAP_SIZE}")
 fi
 
-#### add classpath and java opts entries comes from docker image if defined
+#### add classpath and java opts entries that come from docker image if defined
 if [ "x$CLASSPATH_DEFAULT" != "x" ]; then
     CLASSPATH="${CLASSPATH_DEFAULT}:${CLASSPATH}"
 fi
 
 if [ "x$JAVA_OPTS_DEFAULT" != "x" ]; then
-    JAVA_OPTS="${JAVA_OPTS_DEFAULT} ${JAVA_OPTS}"
+    JAVA_OPTS+=("${JAVA_OPTS_DEFAULT}")
 fi
 
 if [ "x$JET_LICENSE_KEY" != "x" ]; then
-    JAVA_OPTS="$JAVA_OPTS -Dhazelcast.enterprise.license.key=${JET_LICENSE_KEY}"
+    JAVA_OPTS+=("-Dhazelcast.enterprise.license.key=${JET_LICENSE_KEY}")
 fi
 
 if [ "x$JET_LICENCE_KEY" != "x" ]; then
-    JAVA_OPTS="$JAVA_OPTS -Dhazelcast.enterprise.license.key=${JET_LICENCE_KEY}"
+    JAVA_OPTS+=("-Dhazelcast.enterprise.license.key=${JET_LICENCE_KEY}")
 fi
 
 CLASSPATH="$JET_HOME/lib/${hazelcast.jet.artifact}-${project.version}.jar:$CLASSPATH"
-JAVA_OPTS="$JAVA_OPTS -Dhazelcast.config=$JET_HOME/config/hazelcast.xml \
--Djet.home=$JET_HOME \
--Dhazelcast.client.config=$JET_HOME/config/hazelcast-client.xml \
--Dhazelcast.jet.config=$JET_HOME/config/hazelcast-jet.xml"
+JAVA_OPTS+=("-Dhazelcast.config=$JET_HOME/config/hazelcast.xml" \
+"-Djet.home=$JET_HOME" \
+"-Dhazelcast.client.config=$JET_HOME/config/hazelcast-client.xml" \
+"-Dhazelcast.jet.config=$JET_HOME/config/hazelcast-jet.xml")

--- a/hazelcast-jet-distribution/src/main/resources/bin/jet-start.sh
+++ b/hazelcast-jet-distribution/src/main/resources/bin/jet-start.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 SCRIPT_DIR=`dirname "$0"`
 . $SCRIPT_DIR/common.sh
@@ -10,4 +10,4 @@ echo "# CLASSPATH=$CLASSPATH"
 echo "# starting now...."
 echo "########################################"
 
-$JAVA -server $JAVA_OPTS -cp $CLASSPATH com.hazelcast.jet.server.StartServer
+$JAVA -server "${JAVA_OPTS[@]}" -cp "$CLASSPATH" com.hazelcast.jet.server.StartServer

--- a/hazelcast-jet-distribution/src/main/resources/bin/jet-stop.sh
+++ b/hazelcast-jet-distribution/src/main/resources/bin/jet-stop.sh
@@ -1,6 +1,6 @@
-#!/bin/sh
+#!/bin/bash
 
-PIDS=`ps ax | grep com.hazelcast.jet.server.StartServer | grep -v grep | awk '{print $1}'`
+PIDS=$(ps ax | grep com.hazelcast.jet.server.StartServer | grep -v grep | awk '{print $1}')
 
 if [ -z "$PIDS" ]; then
   echo "No Hazelcast Jet server found to stop"

--- a/hazelcast-jet-distribution/src/main/resources/bin/jet.sh
+++ b/hazelcast-jet-distribution/src/main/resources/bin/jet.sh
@@ -1,6 +1,6 @@
-#!/bin/sh
+#!/bin/bash
 
-SCRIPT_DIR=`dirname "$0"`
-. $SCRIPT_DIR/common.sh
+SCRIPT_DIR="$(dirname "$0")"
+. "$SCRIPT_DIR/common.sh"
 
-$JAVA $JAVA_OPTS -cp $CLASSPATH com.hazelcast.jet.server.JetCommandLine "$@"
+$JAVA "${JAVA_OPTS[@]}" -cp "$CLASSPATH" com.hazelcast.jet.server.JetCommandLine "$@"

